msal4j-brokers/src/main/java/com/microsoft/aad/msal4jbrokers/MsalRuntimeBroker.java

```diff
@@ -0,0 +1,192 @@
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4jbrokers;
+
+import com.microsoft.aad.msal4j.IAuthenticationResult;
+import com.microsoft.aad.msal4j.IBroker;
+import com.microsoft.aad.msal4j.InteractiveRequestParameters;
+import com.microsoft.aad.msal4j.PublicClientApplication;
+import com.microsoft.aad.msal4j.SilentParameters;
+import com.microsoft.aad.msal4j.UserNamePasswordParameters;
+import com.microsoft.aad.msal4j.MsalException;
+import com.microsoft.aad.msal4j.AuthenticationErrorCode;
+import com.microsoft.aad.msal4j.IAccount;
+import com.microsoft.azure.javamsalruntime.Account;
+import com.microsoft.azure.javamsalruntime.AuthParameters;
+import com.microsoft.azure.javamsalruntime.AuthResult;
+import com.microsoft.azure.javamsalruntime.MsalInteropException;
+import com.microsoft.azure.javamsalruntime.MsalRuntimeInterop;
+import com.microsoft.azure.javamsalruntime.ReadAccountResult;
+
+import java.net.MalformedURLException;
+import java.util.Collections;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+public class MsalRuntimeBroker implements IBroker {
+    private static MsalRuntimeInterop interop;
+
+    static {
+        try {
+            //MsalRuntimeInterop performs various initialization steps in a similar static block,
+            // so when an MsalRuntimeBroker is created this will cause the interop layer to initialize
+            interop = new MsalRuntimeInterop();
+        } catch (MsalInteropException e) {
+            //TODO: Meaningful error message explaining why MSALRuntime failed to initialize (unsupported platform, error from MSALRuntime, etc.)
+        }
+    }
+
+    @Override
+    public void acquireToken(PublicClientApplication application, SilentParameters parameters, CompletableFuture<IAuthenticationResult> future) {
+        Account accountResult = null;
+
+        //If request has an account ID, MSALRuntime data cached for that account
+        //  try to get account info from MSALRuntime
+        if (parameters.account() != null) {
+            try {
+                accountResult = ((ReadAccountResult) interop.readAccountById(parameters.account().homeAccountId(), application.correlationId()).get()).getAccount();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            } catch (ExecutionException e) {
+                e.printStackTrace();
+            }
+        }
+
+        try (AuthParameters authParameters =
+                     new AuthParameters
+                             .AuthParametersBuilder(application.clientId(),
+                             application.authority(),
+                             parameters.scopes().toString())
+                             .build()) {
+
+            //Account was not cached in MSALRuntime, must perform sign in first to populate account info
+            if (accountResult == null) {
```

Comment:
nit - It's a better practice to check for nulls this way - https://stackoverflow.com/questions/10983573/checking-for-null-what-order